### PR TITLE
Add unit tests for com.zheng.common.util.Paginator

### PR DESCRIPTION
**Pull request by allredj**
I've analysed your codebase and noticed that `com.zheng.common.util.Paginator` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).
Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important in a subsequent PR.

### CODE REVIEW

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import java.lang.reflect.Method;
+import javax.servlet.http.HttpServletRequest;
+
+@RunWith(PowerMockRunner.class)
+public class PaginatorTest {
+
+  @PrepareForTest({Paginator.class, HttpServletRequest.class})
+  @Test
+  public void constructor() throws Exception {
```
Comment (JohnLBergqvist) [inline zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: We don't submit tests for constructors

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    Assert.assertEquals(5, objectUnderTest.getStep());
+    Assert.assertEquals("BAZ", objectUnderTest.getUrl());
+    Assert.assertEquals("1a 2b 3c", objectUnderTest.getQuery());
+    Assert.assertEquals("page", objectUnderTest.getParam());
+    Assert.assertEquals(1, objectUnderTest.getPage());
+    Assert.assertEquals(1, objectUnderTest.getRows());
+  }
+
+  @Test
+  public void setTotal() {
```
Comment (JohnLBergqvist) [inline zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: We don't submit tests for getters & setters.

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    final Paginator objectUnderTest = new Paginator();
+
+    objectUnderTest.setRows(10_001);
+
+    Assert.assertEquals(0L, objectUnderTest.getTotalPage());
+    Assert.assertEquals(10_000, objectUnderTest.getRows());
+  }
+
+  @Test
+  public void initTotalPageDefault() {
```
Comment (JohnLBergqvist) [inline zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: have these tests have their name start like `testInitTotal`...

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
Fix your commit message.

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java (hunk 1 above)
Comment (allredj) [reply zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: This one has interesting side-effects to assert, though.

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java (hunk 2 above)
Comment (allredj) [reply zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: This one has interesting side-effects that ought to be tested.

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java (hunk 3 above)
Comment (allredj) [reply zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: Tests renamed

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java (hunk 1 above)
Comment (JohnLBergqvist) [reply zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: @allredj Run it past @peterschrammel

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java (hunk 2 above)
Comment (JohnLBergqvist) [reply zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: @allredj Run it past @peterschrammel

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java (hunk 1 above)
Comment (paulkennethkent) [reply zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: I think this is interesting enough to submit.

File: zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java (hunk 2 above)
Comment (peterschrammel) [reply zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java]: LGTM

**Review by JohnLBergqvist — APPROVED** [review]

**Pull request closed without merging**